packages/opentelemetry-core/src/context/propagation/TraceContextFormat.ts

```diff
@@ -0,0 +1,117 @@
+// Consider to import const from TraceOptions
+export const DEFAULT_OPTIONS = 0x0;
+const TRACE_PARENT_REGEX = /^[\da-f]{2}-[\da-f]{32}-[\da-f]{16}-[\da-f]{2}$/;
+const VERSION = '00';
+const traceIdFormat = /^[0-9a-f]{32}$/i;
+const spanIdFormat = /^[0-9a-f]{16}$/i;
+const invalidIdFormat = /^0+$/i;
+
+// TODO: Conside to move these interfaces in types.
+export interface HeaderGetter {
+  getHeader(name: string): string | string[] | undefined;
+}
+export interface HeaderSetter {
+  setHeader(name: string, value: string): void;
+}
+
+/**
+ * Propagates {@link SpanContext} through Trace Context format propagation.
+ *
+ * Based on the Trace Context specification:
+ * https://www.w3.org/TR/trace-context/
+ */
+export class TraceContextFormat implements Propagator {
+  inject(spanContext: SpanContext, format: string, carrier: unknown) {
+    const setter: HeaderSetter = carrier as HeaderSetter;
+    const traceParent = `${VERSION}-${spanContext.traceId}-${
+      spanContext.spanId
+    }-0${(spanContext.traceOptions || DEFAULT_OPTIONS).toString(16)}`;
+
+    setter.setHeader(TRACE_PARENT_HEADER, traceParent);
+    if (spanContext.traceState) {
+      // TODO: https://github.com/open-telemetry/opentelemetry-js/pull/57
+      //setter.setHeader(TRACE_STATE_HEADER, spanContext.traceState);
+    }
+  }
+
+  extract(format: string, carrier: unknown): SpanContext | null {
+    const getter: HeaderGetter = carrier as HeaderGetter;
+    const traceParentHeader = getter.getHeader(TRACE_PARENT_HEADER);
+    if (!traceParentHeader) return null;
+    const traceParent = Array.isArray(traceParentHeader)
+      ? traceParentHeader[0]
+      : traceParentHeader;
+    const spanContext = parse(traceParent);
+    if (!spanContext) return null;
+
+    const traceStateHeader = getter.getHeader(TRACE_STATE_HEADER);
+    if (traceStateHeader) {
+      // If more than one `tracestate` header is found, we merge them into a
+      // single header.
+      // TODO: https://github.com/open-telemetry/opentelemetry-js/pull/57
+      spanContext.traceState = Array.isArray(traceStateHeader)
+        ? traceStateHeader.join(',')
+        : traceStateHeader;
+    }
+    return spanContext;
+  }
+}
+
+export function parse(traceParent: string): SpanContext | null {
+  const match = traceParent.match(TRACE_PARENT_REGEX);
+  if (!match) return null;
+  const parts = traceParent.split('-');
+  const [version, traceId, spanId, option] = parts;
+  // tslint:disable-next-line:ban Needed to parse hexadecimal.
+  const traceOptions = parseInt(option, 16);
+
+  if (
+    !isValidVersion(version) ||
+    !isValidTraceId(traceId) ||
+    !isValidSpanId(spanId)
+  ) {
+    return null;
+  }
+
+  return { traceId, spanId, traceOptions };
+}
+
+function isValidVersion(version: string): boolean {
+  if (version !== VERSION) return false;
+  return true;
+}
+
+function isValidTraceId(traceId: string): boolean {
+  if (!traceIdFormat.test(traceId) || invalidIdFormat.test(traceId)) {
```

Comment:
Can this just be `return traceIdFormat.test(traceId) && !invalidIdFormat.test(traceId);`?